attributes.go

```diff
@@ -108,3 +110,36 @@ func statusCodeAttribute(protocol string, serverErr error) attribute.KeyValue {
 	}
 	return codeKey.String("success")
 }
+
+func headerAttributes(protocol, eventType string, metadata http.Header, allowedKeys []string) []attribute.KeyValue {
+	var attributes []attribute.KeyValue
+	for _, allowedKey := range allowedKeys {
+		if val, ok := metadata[allowedKey]; ok {
+			keyValue := attribute.String(
+				formatHeaderAttributes(protocol, eventType, formatHeaderAttributeKey(allowedKey)),
+				formatHeaderAttributeValue(val),
+			)
+			attributes = append(attributes, keyValue)
+		}
+	}
+	return attributes
+}
+
+func formatHeaderAttributes(protocol, eventType, key string) string {
+	return fmt.Sprintf("rpc.%s.%s.metadata.%s", protocol, eventType, key)
+}
+
+func formatHeaderAttributeKey(key string) string {
+	return strings.ReplaceAll(strings.ToLower(key), "-", "_")
+}
+
+func formatHeaderAttributeValue(val []string) string {
```

Comment:
Implemented formatting `rpc.request.metadata.my_custom_metadata_attribute=["1.2.3.4", "1.2.3.5"]` suggested [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md#grpc-request-and-response-metadata)